### ISSUE TITLE
model.drop() not resetting isCreatedInDatabase flag

### ISSUE THREAD

**Issue by SquareMesh** [eveningkid/denodb#56]
Application using db.sync to recreate model in the database.

await db.sync({ drop: true });

Operation works successfully the first time, but running a 2nd time causes failure ""This model has already been initialized." as it believes model still exists in the database even though its deleted.

Discovered property _isCreatedInDatabase is not set to false when drop is called.

Not enough experience/confidence to undertake pull request, but maybe change to drop() method in model.ts akin to suggestion below..

```
 /** Drop a model in the database. */
  static async drop() {
    const dropQuery = this._options.queryBuilder.queryForSchema(this).table(
      this.table,
    ).dropIfExists().toDescription();
    
    return this._options.database.query(dropQuery).then(() => {this._isCreatedInDatabase = false});
  }
```

**Comment by eveningkid** [comment]
Hey,

Yes, this is a mistake, thanks for spotting it :)

I am sure you have experience/confidence to make a PR next time really.

I will just use `async/await` instead but your solution is correct!

Thanks a lot for the suggestion, I will update this right now

**Comment by eveningkid** [comment]
This was fixed with 4488debdda3ffd95fd3ec4bae4916a018e256582 :)

Thanks again

**Comment by SquareMesh** [comment]
Great stuff and thanks for the feedback/advice on solution.
